parmed/modeller/residue.py

```diff
@@ -251,7 +251,10 @@ def __getitem__(self, idx):
                 if atom.name == idx:
                     return atom
             raise IndexError('Atom %s not found in %s' % (idx, self.name))
-        return self.atoms[idx]
+        elif isinstance(idx, (list, tuple)):
+            return [self[key] for key in  idx]
```

Comment:
This gets a few things wrong... First, it should be `hasattr(idx, '__iter__')`, since it could be a generator or other iterable, like numpy array.  Second, it could be a mask array: `if len(idx) == len(self) and all(isinstance(x, bool) for x in idx)`, in which case the result should be `return [self[x] for x in idx if x]`

Also, the tuple could contain multiples of the same index.  The equivalent of what `Structure.__getitem__` does here is:

```Python
    return [self[x] for x in sorted(set(idx))]
```